pkg/autodetect/main.go

```diff
@@ -71,7 +82,7 @@ func (a *autoDetect) Platform() (platform.Platform, error) {
 
 	apiGroups := apiList.Groups
 	for i := 0; i < len(apiGroups); i++ {
-		if apiGroups[i].Name == "route.openshift.io" {
+		if apiGroups[i].Name == "operator.openshift.io" {
```

Comment:
With this change, routes are now recognized under openshift but no longer under k8s, correct?

Reply:
The problem I'm trying to solve is this:
- Detecting if the platform is OpenShift is based on finding the `route.openshift.io` resource.
- The `route.openshift.io` is installed in KiND as part of the E2E tests
- The E2E tests think they are running in OpenShift but that is not true

So, I modified the code to:
- Detect if the platform is OpenShift based on finding another resource (I chose `operator.openshift.io` but can be another one except `route.openshift.io`)
- Add methods to detect if `route.openshift.io` is installed (what is true for OpenShift and Kubernetes clusters where `routes.openshift.io` is installed)

Reply:
ok, i see. Then i would follow the same approach as the Platform detection and use the method to verify if routes are available in the reconciler.